src/ol/render/canvas/hitdetect.js

```diff
@@ -192,5 +193,5 @@ export function hitDetect(pixel, features, imageData) {
       resultFeatures.push(features[i / indexFactor - 1]);
     }
   }
-  return resultFeatures;
+  return /** @type {Array<F>} */ (resultFeatures);
```

Comment:
Not sure why it doesn't work without this cast. Its simply copying features from the input array to the returned array ....